src/hooks/core/chat/useDMMessage.ts

```diff
@@ -0,0 +1,144 @@
+
+      channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
+      channel.myUserId = signedProfileId; // change to use getChannelListInfo
+      channel.targetName = chatName?.name; // change to use getChannelListInfo
+      channel.targetImage = chatName?.image; // change to use getChannelListInfo
+
+      channel.channel = {...channel};
+      console.log('data', {
+        anon_user_info_color_name: anonUserInfoColorName,
+        anon_user_info_color_code: anonUserInfoColorCode,
+        anon_user_info_emoji_code: anonUserInfoEmojiCode,
+        anon_user_info_emoji_name: anonUserInfoEmojiName
+      });
+      const channelList = ChannelList.fromChannelAPI(channel, chatType, channel?.members, {
+        anon_user_info_color_name: anonUserInfoColorName,
+        anon_user_info_color_code: anonUserInfoColorCode,
+        anon_user_info_emoji_code: anonUserInfoEmojiCode,
+        anon_user_info_emoji_name: anonUserInfoEmojiName
+      });
+      await channelList.saveIfLatest(localDb);
+      refresh('channelList', 'chat', 'channelInfo', 'channelMember');
+      return goToChatScreen(channelList);
+    } catch (error) {
+      console.log('sendMessage', error);
+    }
+  };
+
+  const sendMessageDM = async (
+    id,
+    source: 'post' | 'comment',
+    channelCategory: ChannelCategory
+  ) => {
+    const initChat =
+      channelCategory === 'ANONYMOUS'
+        ? await initChatFromPostAnon({source, id})
+        : await initChatFromPost({source, id});
+
+    const builtChannelData = {
+      better_channel_member: initChat.data.better_channel_members,
+      members: initChat?.data.members
+    };
+
+    const channel = {
+      ...initChat?.data?.channel,
+      better_channel_members: initChat?.data?.better_channel_members,
+      members: initChat?.data?.better_channel_members
+    };
+
+    if (!channel?.members?.length) {
+      console.info('no members');
+      return;
+    }
+
+    if (channel?.type === 'topics') {
+      console.info('type is topics');
+      return;
+    }
+
+    try {
+      await Promise.all(
+        (channel?.members || []).map(async (member) => {
+          const userMember = UserSchema.fromMemberWebsocketObject(member, channel?.id);
+          const memberSchema = ChannelListMemberSchema.fromWebsocketObject(
+            channel?.id,
+            uuid(),
+            member // change to use getChannelListInfo
+          );
+          await userMember.saveOrUpdateIfExists(localDb);
+          await memberSchema.save(localDb);
+        })
+      );
+
+      await Promise.all(
+        (channel?.messages || []).map(async (message) => {
+          if (message?.type === 'deleted') return;
+          const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
+          await chat.save(localDb);
+        })
+      );
+
+      await saveChannelList(channel, channelCategory, builtChannelData);
+    } catch (e) {
+      console.log('error on saveChannelData', e);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similarly, in the `sendMessageDM` function, you're catching errors but only logging them to the console. It would be more helpful to propagate these errors up to the caller so they can handle them appropriately. 

```diff
    } catch (e) {
      console.log('error on saveChannelData', e);
+      throw e;
    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->